cli/fathom_web/commands/vectorize.py

```diff
@@ -0,0 +1,312 @@
+    # Windows, if the user is using cygwin, this command returns a cygwin path,
+    # so we need to transform this into a real Windows path.
+    # See: https://stackoverflow.com/questions/39085380/how-can-i-suppress-terminate-batch-job-y-n-confirmation-in-powershell
+    # TODO: Better error message for not having which or yarn
+    yarn_dir = subprocess.run(['which', 'yarn'], capture_output=True).stdout.decode().strip()[:-4]
+    if 'cygdrive' in yarn_dir:
+        yarn_dir = subprocess.run(['cygpath', '-w', yarn_dir], capture_output=True).stdout.decode().strip()
+    # TODO: Better error message for not having node or rollup
+    subprocess.run(['node', f'{yarn_dir}/yarn.js', '--cwd', fathom_trainees_dir, 'build'], capture_output=True)
+
+    fathom_trainees = create_xpi_for(pathlib.Path(fathom_trainees_dir) / 'addon', 'fathom-trainees', temp_dir)
+    print('Done')
+    return fathom_fox, fathom_trainees
+
+
+def create_xpi_for(directory, name, dest_dir):
+    """Create an .xpi archive for a directory and returns its absolute path."""
+    xpi_path = pathlib.Path(dest_dir.name) / f'{name}.xpi'
+    with zipfile.ZipFile(xpi_path, 'w', compression=zipfile.ZIP_DEFLATED) as xpi:
+        for file in directory.rglob('*'):
+            xpi.write(file, file.relative_to(directory))
+    return str(xpi_path.absolute())
+
+
+def run_file_server(samples_directory):
+    """
+    Create a local HTTP server for the samples.
+
+    We return both the server and the thread to handle a graceful or forced
+    shutdown.
+    """
+    print('Starting HTTP file server...', end='', flush=True)
+    RequestHandler = partial(SilentRequestHandler, directory=samples_directory)
+    server = HTTPServer(('localhost', 8000), RequestHandler)
+    server_thread = Thread(target=server.serve_forever)
+    server_thread.start()
+    print('Done')
+    return server, server_thread
+
+
+def configure_firefox(fathom_fox, fathom_trainees, output_directory, show_browser, temp_dir):
+    """
+    Configures and launches Firefox to run the vectorizer with.
+
+    Sets headless mode, sets the download directory to the desired output
+    directory, turns off page caching, and installs FathomFox and Fathom
+    Trainees.
+
+    We return the webdriver object, and the process IDs for both the Firefox
+    process and the geckoview driver process so we can shutdown either
+    gracefully or ungracefully.
+    """
+    print('Configuring Firefox...', end='', flush=True)
+    options = webdriver.FirefoxOptions()
+    options.headless = not show_browser
+
+    profile = webdriver.FirefoxProfile()
+    profile.set_preference('browser.download.folderList', 2)
+    profile.set_preference('browser.download.dir', str(pathlib.Path(output_directory).absolute()))
+    profile.set_preference('browser.cache.disk.enable', False)
+    profile.set_preference('browser.cache.memory.enable', False)
+    profile.set_preference('browser.cache.offline.enable', False)
```

Comment:
`browser.cache.disk.enable` will store pages on the hard disk. `browser.cache.memory.enable` will store pages in memory. `browser.cache.offline.enable` will store `<link rel="offline-resource">`s in some cache.

The disk version seems most important to me. The memory version seems debatable since each page is only loaded once during the `fathom-vectorize` run. The offline version is probably unnecessary, but I don't see a problem leaving it.

I don't know if there is a better source for what these do, but I used:
http://kb.mozillazine.org/Browser.cache.disk.enable
http://kb.mozillazine.org/Browser.cache.memory.enable
http://kb.mozillazine.org/Browser.cache.offline.enable

Reply:
I don't think there's a great way other than looking at the blame in [SearchFox](https://searchfox.org/), which is a searchable directory of the tip of mozilla-central's source code, for where the pref is initialized and hoping you can track down the bug that introduced it in Bugzilla for more context. I didn't have much luck doing that though for at least the first pref.

I have never heard of MozillaZine! I noticed that was last updated in 2014, and was created by some community members. Could be a place to look, but I would take it with a grain of salt generally. I doubt these particular prefs have changed much if at all since 2014 in terms of what they control.